File: leetcode/js/decode-xored-array.js
/**
 * @param {number[]} encoded
 * @param {number} first
 * @return {number[]}
 */
var decode = function (encoded, first) {
  const arr = [first];

  for (const num of encoded) {
    arr.push(arr.at(-1) ^ num);
  }

  return arr;
};

File: leetcode/rust/decode-xored-array.rs
impl Solution {
    pub fn decode(encoded: Vec<i32>, first: i32) -> Vec<i32> {
        let mut arr = Vec::from([first]);

        for (idx, &num) in encoded.iter().enumerate() {
            arr.push(arr[idx] ^ num);
        }

        arr
    }
}
